src/pubsub/ua_pubsub_reader.c

```diff
@@ -1865,10 +1892,9 @@ decodeNetworkMessage(UA_Server *server,
     UA_dump_hex_pkg(buffer->data, buffer->length);
 #endif
 
-    UA_StatusCode rv = UA_NetworkMessage_decodeHeaders(
-        buffer, currentPosition, currentNetworkMessage);
-    UA_CHECK_STATUS_ERROR(rv, return rv, logger, UA_LOGCATEGORY_SERVER,
-                          "PubSub receive. decoding headers failed");
+    /* Decode the headers and decrypt the NM */
+    UA_StatusCode rv = decodeHeadersAndDecryptNM(server, connection->readerGroups.lh_first, buffer);
```

Comment:
This change is wrong.

- The same code-path calls `verifyAndDecryptNetworkMessage` below. So the decrypt routine is called twice on the same buffer.
- I think we need to have currentNetworkMessage updated to contain the decrypted headers. This is not done with the new code.

Just keep the old code here.